File: frontend/lib/passwordPolicy.ts
/** Mirrors backend `src/utils/passwordPolicy.ts`. */
export const PASSWORD_POLICY_HINT =
  "At least 6 characters, with 1 uppercase letter, 1 lowercase letter, and a number.";

export function passwordPolicyIssue(password: string): string | null {
  if (password.length < 6) return "Minimum 6 characters";
  if (!/[A-Z]/.test(password)) return "Must contain at least one uppercase letter";
  if (!/[a-z]/.test(password)) return "Must contain at least one lowercase letter";
  if (!/[0-9]/.test(password)) return "Must contain at least one number";
  if (!/[A-Za-z]/.test(password)) return "Must contain letters";
  return null;
}

File: nb_crm_flutter/lib/core/utils/password_policy.dart
/// Mirrors backend `passwordPolicy.ts`.
const _upper = '[A-Z]';
const _lower = '[a-z]';
const _digit = '[0-9]';

const passwordPolicyHint =
    'At least 6 characters, with 1 uppercase letter, 1 lowercase letter, and a number.';

String? passwordPolicyIssue(String password) {
  if (password.length < 6) return 'Minimum 6 characters';
  if (!RegExp(_upper).hasMatch(password)) {
    return 'Must contain at least one uppercase letter';
  }
  if (!RegExp(_lower).hasMatch(password)) {
    return 'Must contain at least one lowercase letter';
  }
  if (!RegExp(_digit).hasMatch(password)) {
    return 'Must contain at least one number';
  }
  if (!RegExp(r'[A-Za-z]').hasMatch(password)) {
    return 'Must contain letters';
  }
  return null;
}

bool passwordMeetsPolicy(String password) =>
    passwordPolicyIssue(password) == null;

String? validateNewPassword(String? value) {
  if (value == null || value.isEmpty) {
    return 'New password is required';
  }
  return passwordPolicyIssue(value);
}
